controllers/terraform_controller.go

```diff
@@ -1597,74 +1620,106 @@ func (r *TerraformReconciler) reconcileRunnerSecret(ctx context.Context, terrafo
 	return nil
 }
 
-func (r *TerraformReconciler) reconcileRunnerPod(ctx context.Context, terraform *infrav1.Terraform) error {
-	runner := &corev1.Pod{
+func (r *TerraformReconciler) reconcileRunnerPod(ctx context.Context, terraform infrav1.Terraform) (string, error) {
+	podNamespace := terraform.Namespace
+	podName := fmt.Sprintf("%s-runner", terraform.Name)
+	runnerPod := corev1.Pod{
 		ObjectMeta: metav1.ObjectMeta{
-			Namespace: terraform.Namespace,
-			Name:      fmt.Sprintf("%s-runner", terraform.Name),
+			Namespace: podNamespace,
+			Name:      podName,
 		},
 	}
 
-	if err := r.Get(ctx, client.ObjectKeyFromObject(runner), runner); err != nil {
+	runnerPodKey := client.ObjectKeyFromObject(&runnerPod)
+	if err := r.Get(ctx, runnerPodKey, &runnerPod); err != nil {
 		if !apierrors.IsNotFound(err) {
-			return err
+			return "", err
 		}
-		runner.Spec = runnerPodSpec(infrav1.RunnerTLSSecretName)
-		if err := r.Create(ctx, runner); err != nil {
-			return err
+		runnerPod.Spec = runnerPodSpec()
+		if err := r.Create(ctx, &runnerPod); err != nil {
+			return "", err
 		}
 	}
 
-	return nil
+	// TODO make it configurable
+	const maxRetry = 5
+	retry := 1
+	for {
+		if err := r.Get(ctx, runnerPodKey, &runnerPod); err != nil {
+			return "", err
+		}
+
+		podIp := runnerPod.Status.PodIP
+		if podIp != "" {
+			return podIp, nil
+		}
+
+		retry++
+		if retry == maxRetry {
+			return "", fmt.Errorf("wait for max retry reached")
+		}
+
+		// TODO make it configurable
+		time.Sleep(2 * time.Second)
+	}
+
 }
 
-func runnerPodSpec(tlsSecretName string) corev1.PodSpec {
+func getRunnerPodImage() string {
+	runnerPodImage := os.Getenv("RUNNER_POD_IMAGE")
+	if runnerPodImage == "" {
+		runnerPodImage = "ghcr.io/weaveworks/tf-runner:latest"
```

Comment:
Should we pin to an image tag that matches the controller image tag?

Reply:
We can pin it via the env here for every release: https://github.com/weaveworks/tf-controller/pull/101/files#diff-ce0d597a9e5f20a8fe61c3ba5e6185a6b90cfef2c00bb5b5ef3e97b15b79f480R49

Making it as an env so that it's flexible during the development and test.

Also, it allows hotfix when something unexpected breaks the users.